sdk/python/kfp/components/structures.py

```diff
@@ -585,21 +605,81 @@ def from_v1_component_spec(
             'env': container['env']
         })
 
+        inputs = {}
+        for spec in component_dict.get('inputs', []):
+            type_ = spec.get('type')
+
+            if isinstance(type_, str) and type_.lower(
+            ) in type_utils._PARAMETER_TYPES_MAPPING:
+                schema_version = None
+
+            elif isinstance(type_, str) and re.match(
+                    type_utils._GOOGLE_TYPES_PATTERN, type_):
+                schema_version = type_utils._GOOGLE_TYPES_VERSION
+
+            elif isinstance(type_, str) and type_.lower(
+            ) in type_utils._ARTIFACT_CLASSES_MAPPING:
+                artifact_class = type_utils._ARTIFACT_CLASSES_MAPPING[
+                    type_.lower()]
+                type_ = artifact_class.schema_title
+                schema_version = artifact_class.schema_version
+
+            elif type_ == 'PipelineTaskFinalStatus':
+                schema_version = type_utils.DEFAULT_ARTIFACT_SCHEMA_VERSION
+
+            elif type_ is None or isinstance(type_, dict) or type_.lower(
+            ) not in type_utils._ARTIFACT_CLASSES_MAPPING:
+                type_ = artifact_types.Artifact.schema_title
+                schema_version = artifact_types.Artifact.schema_version
+
+            else:
+                schema_version = type_utils.DEFAULT_ARTIFACT_SCHEMA_VERSION
+
+            default = spec.get('default', None)
+            inputs[utils.sanitize_input_name(spec['name'])] = InputSpec(
+                type=type_,
+                default=default,
+                schema_version=schema_version,
+            )
+
+        outputs = {}
+        for spec in component_dict.get('outputs', []):
+            type_ = spec.get('type')
+
+            if isinstance(type_, str) and type_.lower(
+            ) in type_utils._PARAMETER_TYPES_MAPPING:
+                schema_version = None
+
+            elif isinstance(type_, str) and re.match(
+                    type_utils._GOOGLE_TYPES_PATTERN, type_):
+                schema_version = type_utils._GOOGLE_TYPES_VERSION
+
+            elif isinstance(type_, str) and type_.lower(
+            ) in type_utils._ARTIFACT_CLASSES_MAPPING:
+                artifact_class = type_utils._ARTIFACT_CLASSES_MAPPING[
+                    type_.lower()]
+                type_ = artifact_class.schema_title
+                schema_version = artifact_class.schema_version
+
+            elif type_ is None or isinstance(type_, dict) or type_.lower(
+            ) not in type_utils._ARTIFACT_CLASSES_MAPPING:
+                type_ = artifact_types.Artifact.schema_title
+                schema_version = artifact_types.Artifact.schema_version
+
+            else:
+                schema_version = type_utils.DEFAULT_ARTIFACT_SCHEMA_VERSION
+            outputs[utils.sanitize_input_name(spec['name'])] = OutputSpec(
+                type=type_,
+                schema_version=schema_version,
+            )
+
```

Comment:
Update the v1 deserialization logic to have more intelligent fallback to the generic Artifact class.